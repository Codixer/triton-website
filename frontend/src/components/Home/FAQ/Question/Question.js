import React, { Component } from "react";
import styles from "./Question.scss";
import OpenSansTypography from "../../../OpenSansTypography/OpenSansTypography";
import { Typography } from "@rmwc/typography";
import "@material/typography/dist/mdc.typography.css";
import { Elevation } from "@rmwc/elevation";
import "@material/elevation/dist/mdc.elevation.css";
import { GridCell } from "@rmwc/grid";
import "@material/layout-grid/dist/mdc.layout-grid.css";
import commentQuestionIcon from "../../../../img/icons/comment-question.svg";
import { Icon } from "@rmwc/icon";
import "@rmwc/icon/icon.css";

class Question extends Component {
  render() {
    return (
      <GridCell span="4">
        <Elevation z={8} className={styles.question}>
          <div className={styles.headerDiv}>
            <Icon icon={commentQuestionIcon} />
            <OpenSansTypography use="headline6">{this.props.title}</OpenSansTypography>
          </div>
          <br />
          <Typography use="body1">{this.props.children}</Typography>
        </Elevation>
      </GridCell>
    );
  }
}

export default Question;
