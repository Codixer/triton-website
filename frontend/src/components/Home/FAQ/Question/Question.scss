:local(.question) {
  height: 100%;
  border-radius: 15px;
  padding: 10px;
  box-sizing: border-box;
}

:local(.headerDiv) {
  display: flex;
  align-items: center;
  justify-content: flex-start;
  .rmwc-icon {
    padding-right: 10px;
  }
}
