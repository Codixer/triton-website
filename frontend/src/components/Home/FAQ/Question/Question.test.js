import React from "react";
import { shallow } from "enzyme";
import Question from "./Question";

describe("Home/FAQ/Question Component", () => {
  it("matches snapshot", () => {
    let wrapper = shallow(
      <Question title="This is a test title!">This is a test answer!</Question>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
