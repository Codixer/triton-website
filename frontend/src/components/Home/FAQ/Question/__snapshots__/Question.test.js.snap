// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home/FAQ/Question Component matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Question
    title="This is a test title!"
  >
    This is a test answer!
  </Question>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <Elevation
        z={8}
      >
        <div>
          <Icon
            icon="comment-question.svg"
          />
          <OpenSansTypography
            use="headline6"
          >
            This is a test title!
          </OpenSansTypography>
        </div>
        <br />
        <Typography
          use="body1"
        >
          This is a test answer!
        </Typography>
      </Elevation>,
      "span": "4",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div>
            <Icon
              icon="comment-question.svg"
            />
            <OpenSansTypography
              use="headline6"
            >
              This is a test title!
            </OpenSansTypography>
          </div>,
          <br />,
          <Typography
            use="body1"
          >
            This is a test answer!
          </Typography>,
        ],
        "className": undefined,
        "z": 8,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Icon
                icon="comment-question.svg"
              />,
              <OpenSansTypography
                use="headline6"
              >
                This is a test title!
              </OpenSansTypography>,
            ],
            "className": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "icon": "comment-question.svg",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "This is a test title!",
                "use": "headline6",
              },
              "ref": null,
              "rendered": "This is a test title!",
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "This is a test answer!",
            "use": "body1",
          },
          "ref": null,
          "rendered": "This is a test answer!",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Elevation
          z={8}
        >
          <div>
            <Icon
              icon="comment-question.svg"
            />
            <OpenSansTypography
              use="headline6"
            >
              This is a test title!
            </OpenSansTypography>
          </div>
          <br />
          <Typography
            use="body1"
          >
            This is a test answer!
          </Typography>
        </Elevation>,
        "span": "4",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <div>
              <Icon
                icon="comment-question.svg"
              />
              <OpenSansTypography
                use="headline6"
              >
                This is a test title!
              </OpenSansTypography>
            </div>,
            <br />,
            <Typography
              use="body1"
            >
              This is a test answer!
            </Typography>,
          ],
          "className": undefined,
          "z": 8,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Icon
                  icon="comment-question.svg"
                />,
                <OpenSansTypography
                  use="headline6"
                >
                  This is a test title!
                </OpenSansTypography>,
              ],
              "className": undefined,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "icon": "comment-question.svg",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "This is a test title!",
                  "use": "headline6",
                },
                "ref": null,
                "rendered": "This is a test title!",
                "type": [Function],
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "This is a test answer!",
              "use": "body1",
            },
            "ref": null,
            "rendered": "This is a test answer!",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
